notebooks/performance_monitoring.ipynb

```diff
@@ -0,0 +1,487 @@
+    "            linewidth=1.5,\n",
+    "        )\n",
+    "\n",
+    "plt.title(\"Prediction Distributions by Month (Last 12 Months)\", fontsize=12)\n",
+    "plt.xlabel(\"Prediction Value\", fontsize=10)\n",
+    "plt.ylabel(\"Density\", fontsize=10)\n",
+    "legend = plt.legend(title=\"Month\", fontsize=8)\n",
+    "legend.set_title(\"Month\", prop={\"size\": 9})\n",
+    "plt.grid(axis=\"y\", linestyle=\"--\", alpha=0.6)\n",
+    "plt.tick_params(axis=\"x\", labelsize=8)\n",
+    "plt.tick_params(axis=\"y\", labelsize=8)\n",
+    "plt.tight_layout()\n",
+    "plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "11",
+   "metadata": {},
+   "source": [
+    "## Results performance monitoring"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "12",
+   "metadata": {},
+   "source": [
+    "### Load SQL data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "13",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Make new export and save to poliafspraken_no_show.csv\n",
+    "# export_data()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "14",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "appointments_df = load_appointment_csv(\"../data/raw/poliafspraken_no_show.csv\")\n",
+    "appointments_df = process_appointments(appointments_df, CLINIC_CONFIG)\n",
+    "appointments_df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "15",
+   "metadata": {},
+   "source": [
+    "### ROC and AUC scores"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "16",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predicted_patients = (\n",
+    "    df_not_called.groupby(\"month\")[\"appointment_id\"]\n",
+    "    .apply(list)\n",
+    "    .reset_index(name=\"appointment_ids\")\n",
+    ")\n",
+    "\n",
+    "all_postalcodes = process_postal_codes(\"../data/raw/NL.txt\")\n",
+    "with open(\"../output/models/no_show_model_cv.pickle\", \"rb\") as f:\n",
+    "    model = pickle.load(f)\n",
+    "\n",
+    "roc_data = []\n",
+    "\n",
+    "for month, appointment_ids in zip(\n",
+    "    predicted_patients[\"month\"], predicted_patients[\"appointment_ids\"], strict=True\n",
+    "):\n",
+    "    if month <= pd.Period(\"2024-12\", freq=\"M\"):\n",
+    "        continue\n",
+    "\n",
+    "    appointment_ids = [int(app_id) for app_id in appointment_ids]\n",
+    "    filtered_df = appointments_df[\n",
+    "        appointments_df[\"APP_ID\"].isin(appointment_ids)\n",
+    "    ].copy()\n",
+    "    print(f\"Processing month: {month}, Appointments: {len(filtered_df)}\")\n",
+    "\n",
+    "    if filtered_df.empty:\n",
+    "        print(f\"No matching appointments found in data for month: {month}, skipping...\")\n",
+    "        continue\n",
+    "\n",
+    "    appointments_features = create_features(filtered_df, all_postalcodes).pipe(\n",
```

Comment:
This won't work unfortunately, since we only use appointments in that month to create features, but normally the featuretable uses the data of the past x years to create features, so these features will be missing a lot of older information about earlier no-shows